src/nvim/tui/tui.c

```diff
@@ -844,7 +844,16 @@ static void fix_terminfo(TUIData *data)
   }
 
   if (STARTS_WITH(term, "xterm") || STARTS_WITH(term, "rxvt")) {
-    unibi_set_if_empty(ut, unibi_cursor_normal, "\x1b[?25h");
+    const char *normal = unibi_get_str(ut, unibi_cursor_normal);
+    if (!normal) {
+      unibi_set_str(ut, unibi_cursor_normal, "\x1b[?25h");
+    } else if (STARTS_WITH(normal, "\x1b[?12l")) {
```

Comment:
@justinmk currently a rather simple test on the `$TERM` environment variable, where the xterm driver only claims the terminal if the name starts with the string `xterm`, and leaves it to the TI driver for anything else:
http://bazaar.launchpad.net/~leonerd/libtickit/trunk/view/head:/src/termdriver-xterm.c#L590

A better test might be to simply ask it if it's an xterm, but that requires more interesting parsing of replies in from the terminal so it gets added to the ever-growing pile of "things that require termkey integrated into tickit".